src/java.desktop/share/classes/javax/swing/plaf/basic/BasicSplitPaneDivider.java

```diff
@@ -360,6 +360,20 @@ else if (e.getPropertyName() == JSplitPane.
         }
     }
 
+    /**
+     * {@inheritDoc}
+     * @param enabled {@inheritDoc}
+     */
+    @Override
+    public void setEnabled(boolean enabled) {
+        if (splitPane.isOneTouchExpandable() &&
+                rightButton != null &&
+                leftButton != null) {
+
+            rightButton.setEnabled(enabled);
+            leftButton.setEnabled(enabled);
+        }
+    }
```

Comment:
Is it possible to override `isEnabled` in `rightButton` and `leftButton` so that it returns the state of `JSplitPane`?

Alternatively, the buttons could install a `PropertyChangeListener` for `"enabled"` property and align their state to the host split pane.

Reply:
Right now, the buttons are always enabled and rendered when isOneTouchExpandable is enabled without any scope to disable...We need this code anyway to set/reset buttons state..I guess on top of this code, we can add isEnabled if it is needed but for this issue, isEnabled is not needed..